arcgis-ios-sdk-samples/Layers/Apply mosaic rule to rasters/ApplyMosaicRuleToRastersViewController.swift

```diff
@@ -0,0 +1,132 @@
+        centerRule.mosaicOperation = .blend
+        
+        // A mosaic rule object with byAttribute method and sort on "OBJECTID" field of the service.
+        let byAttributeRule = AGSMosaicRule()
+        byAttributeRule.mosaicMethod = .attribute
+        byAttributeRule.sortField = "OBJECTID"
+        
+        // A mosaic rule object with lockRaster method and locks 3 image rasters.
+        let lockRasterRule = AGSMosaicRule()
+        lockRasterRule.mosaicMethod = .lockRaster
+        lockRasterRule.lockRasterIDs = [1, 7, 12]
+        
+        return ["None": noneRule,
+                "NorthWest": northWestRule,
+                "Center": centerRule,
+                "ByAttribute": byAttributeRule,
+                "LockRaster": lockRasterRule]
+    }()
+    
+    // MARK: Methods
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        // The URL of an image service that supports mosaic rules.
+        let imageServiceURL = URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/amberg_germany/ImageServer")!
+        // Create an image service raster from an online raster service.
+        imageServiceRaster = AGSImageServiceRaster(url: imageServiceURL)
+        // Check if a mosaic rule exists. If not, create one.
+        if imageServiceRaster.mosaicRule == nil {
+            imageServiceRaster.mosaicRule = AGSMosaicRule()
+        }
+        // Create a raster layer.
+        let rasterLayer = AGSRasterLayer(raster: imageServiceRaster)
+        let map = AGSMap(basemap: .topographicVector())
+        // Add raster layer as an operational layer to the map.
+        map.operationalLayers.add(rasterLayer)
+        rasterLayer.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // When loaded, set map view's viewpoint to the image service raster's center.
+                if let center = self.imageServiceRaster.serviceInfo?.fullExtent?.center {
+                    self.mapView.setViewpoint(AGSViewpoint(center: center, scale: 25000.0))
+                }
+                self.setStatus(message: "Image service raster loaded.")
+            }
+        }
+        return map
+    }
+    
+    // MARK: UI
+    
+    func setStatus(message: String) {
+        statusLabel.text = message
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func chooseMosaicRule(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose a mosaic rule to apply to the image service.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        mosaicRulePairs.forEach { name, rule in
+            let action = UIAlertAction(title: name, style: .default) { _ in
+                self.setStatus(message: "\(name) selected.")
```

Comment:
Is it possible to update the status to indicate when the mosaic rule has finished loading? At first, it was hard to tell whether or not the rule had been applied or if it was finished loading.

Reply:
I think we mentioned this in last few review mtgs, and there is no (obvious) way to tell if the rasters are displayed, even if they are loaded. Trevor also mentioned this.

Would you mind to give it a try? refer to "display drawing status" and "display layer view state" samples.

Currently the data is too large (>500MB) to load, so it looks like irresponsive. The current mitigation is to make the rasters smaller in size/resolution. But it still doesn't solve the loading status issue. I'll reconfirm with @TADraeseke

Reply:
Issue logged in runtime/common. 

Meanwhile, I feel that the lack of new dataset is still blocking this PR, as we cannot take a screenshot until it is updated. @vquach2404 Do you think it's OK to merge this PR as-is without the latest screenshot?

Reply:
@vquach2404 I've made some more changes to the sample to show the loading status.

Reply:
Ahh I'm sorry. I noticed that yesterday I didn't push the latest changes! Now it is ready for another review! 😅